walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,63 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> Request_attendees = request.getAttendees();
+    List<TimeRange> res = new LinkedList<>();
+    res.add(TimeRange.WHOLE_DAY);
+    for (Event e : events) {
+      Set<String> Attendees = e.getAttendees();
+      boolean flag = false;
+      for (String a : Attendees) {
+        if (Request_attendees.contains(a)) {
+          flag = true;
+          break;
+        }
+      }
+      if (!flag)
+        continue;
+      TimeRange bad = e.getWhen();
+      ListIterator it = res.listIterator();
+      while (it.hasNext()) {
+        TimeRange t = (TimeRange) it.next();
```

Comment:
pls avoiding using such var name unless it's well-known (e.g i in a for loop), because such name is not descriptive.